src/media/local-stream.ts

```diff
@@ -141,55 +144,80 @@ abstract class _LocalStream extends Stream {
   /**
    * Adds an effect to a local stream.
    *
-   * @param name - The name of the effect.
    * @param effect - The effect to add.
    */
-  async addEffect(name: string, effect: TrackEffect): Promise<void> {
-    // Load the effect
-    this.loadingEffects.set(name, effect);
-    const outputTrack = await effect.load(this.outputTrack);
+  async addEffect(effect: TrackEffect): Promise<void> {
+    // Check if the effect has already been added.
+    if (this.effects.includes(effect)) {
+      throw new WebrtcCoreError(
+        WebrtcCoreErrorType.ADD_EFFECT_FAILED,
+        `Effect ${effect.id} has already been added to this stream.`
+      );
+    }
 
-    // Check that the loaded effect is the latest one and dispose if not
-    if (effect !== this.loadingEffects.get(name)) {
+    // Load the effect. Because loading is asynchronous, keep track of the loading effects.
+    this.loadingEffects.set(effect.kind, effect);
+    await effect.load(this.outputTrack);
+
+    // After loading, check whether or not we still want to use this effect. If another effect of
+    // the same kind was added while this effect was loading, we only want to use the latest effect,
+    // so dispose this one. If the effects list was cleared while this effect was loading, also
+    // dispose it.
```

Comment:
It feels a bit strange that we'll allow overriding an affect of the same type that was added previously _if_ it didn't finish loading yet, but we error on adding an effect of the same type after it's been loaded.  I think I'd expect that subsequent additions of the same type of effect (without removing the previous one) should either always work or always fail?

Reply:
The previous behavior was to override the effect if it didn't finish loading, so I think we should keep that behavior (since the client does depend on it). We didn't handle adding a duplicate effect after it has loaded previously, so I can change it so that it will also override instead of error.